src/apps/vpn/controller.cpp

```diff
@@ -564,6 +570,8 @@ void Controller::handshakeTimeout() {
 
 void Controller::disconnected() {
   logger.debug() << "Disconnected from state:" << m_state;
+  mozilla::glean::session::session_end.set();
+  mozilla::glean_pings::Vpnsession.submit();
```

Comment:
Why is this submit here? I would prefer to have the scheduling implementation on a separate PR.